coderd/audit/request.go

```diff
@@ -154,7 +162,7 @@ func InitRequest[T Auditable](w http.ResponseWriter, p *RequestParams) (*Request
 		auditLog := database.AuditLog{
 			ID:               uuid.New(),
 			Time:             database.Now(),
-			UserID:           httpmw.APIKey(p.Request).UserID,
+			UserID:           uuid.Nil,
```

Comment:
This obviously won't give us what we want. But `httpmw.APIKey(p.Request).UserID` always returns error `developer error: ExtractAPIKey middleware not provided` when auditing the API Key resource on login (perhaps because the cookie isn't set when its called?).
What would be the best way to get User ID when auditing logins? One thought is I could grab the User ID off of the API Key itself; however, this won't work for failed logins where an API Key isn't generated but the user is still recognized. Also it just seems messy. Is there a better way?

Reply:
Oh lol, I should have read down here first.

You are right, this is a bit messy. You can do this:

```golang
		userID := uuid.Nil
		key, ok := httpmw.APIKeyOptional(p.Request)
		if ok {
			userID = key.UserID
		}

		// ...
		auditLog := database.AuditLog{
			UserID:           userID,
```

There still remains the problems for other requests that do not use APIkeys at the start of the request.

Reply:
I think we should bring in @coadler about how to attach the user ID from `postLogin` to an audit log.

Reply:
I think what we could do here is have the function returned by `InitRequest` accept a struct that would allow you to specify an override of the user id. What do you think?

Reply:
@coadler I was thinking something similar and like that idea. Idk if it would be a struct or a function. Either works for me, but some way to modify fields after the init would be good.

Reply:
Yeah, actually we could just add a user id to `RequestParams` and use that if provided. That sounds better